storage/src/mmr/mod.rs

```diff
@@ -0,0 +1,196 @@
+mod mem;
+pub use mem::{verify_proof, verify_range_proof, InMemoryMMR};
+
+use sha2::{Digest, Sha256};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Hash<const N: usize>([u8; N]);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// A Proof contains the information necessary for proving the inclusion of an element, or some
+/// range of elements, in the MMR.
+pub struct Proof<const N: usize> {
+    sz: u64, // total # of nodes in the MMR
```

Comment:
nit: elsewhere in the codebase, we use full variable names (i.e. `size` rather than `sz`) to avoid confusion.

while it is slightly longer, I find it makes it easier to onboard new folks to the codebase (other DB projects I've played with are a maze of shorthand and it makes me feel a lot less comfortable making changes).

Reply:
done